### ISSUE TITLE
too easy to use the wrong jdbc driver with a slick driver

### ISSUE THREAD

**Issue by dnvriend** [slick/slick#1693]
The following code generates the incorrect BLOB type, but expected BYTEA:

[Test class](https://github.com/dnvriend/slick-3.2.0-test/blob/master/src/test/scala/com/github/dnvriend/journal/H2JournalQueriesTest.scala#L7):
```
abstract class JournalDDLTest(val p: JdbcProfile) extends TestSpec {
  import akkaPersistenceRepository._
  import p.api._

  "DDL" should "generate create table statement" in {

    val ddl = JournalTable.schema.createStatements.toList
    pprint.pprintln(ddl)
  }
}

class PGJournalDDLTest extends JournalDDLTest(slick.jdbc.PostgresProfile)
```

[Table definition](https://github.com/dnvriend/slick-3.2.0-test/blob/master/src/main/scala/com/github/dnvriend/AkkaPersistenceRepository.scala) is:

```
  class Journal(tag: Tag) extends Table[JournalRow](tag, "JDBC_JOURNAL") {
    def * : ProvenShape[JournalRow] = (ordering, deleted, persistenceId, sequenceNumber, message, tags) <> (JournalRow.tupled, JournalRow.unapply)

    val ordering: Rep[Long] = column[Long]("ordering", O.AutoInc)
    val persistenceId: Rep[String] = column[String]("persistence_id", O.Length(255, varying = true))
    val sequenceNumber: Rep[Long] = column[Long]("sequence_nr")
    val deleted: Rep[Boolean] = column[Boolean]("deleted", O.Default(false))
    val tags: Rep[Option[String]] = column[Option[String]]("tags", O.Length(255, varying = true))
    val message = column[Array[Byte]]("message")
    val pk: PrimaryKey = primaryKey("journal_pk", (persistenceId, sequenceNumber))
  }

  lazy val JournalTable = new TableQuery(tag => new Journal(tag))
```

The resulting SQL is:

```
CREATE TABLE "JDBC_JOURNAL" (
  "ordering"       BIGSERIAL             NOT NULL,
  "deleted"        BOOLEAN DEFAULT FALSE NOT NULL,
  "persistence_id" VARCHAR(255)          NOT NULL,
  "sequence_nr"    BIGINT                NOT NULL,
  "message"        BLOB                  NOT NULL,
  "tags"           VARCHAR(255)
);
```

**Comment by cvogt** [comment]
When you says *expected BYTEA* is there a crash or bug or is this simply a more reasonable choice in your opinion? @tminglei do you have an opinion here?

**Comment by cvogt** [comment]
(the labels are a first estimation, please indicate if they are innacurate)

**Comment by dnvriend** [comment]
The postgres database only supports the keyword `BYTEA` for the BLOB type as can be read from the [postgres datatype-binary documentation](https://www.postgresql.org/docs/9.6/static/datatype-binary.html) but the SQL that the PostgresProfile 'driver' creates for the scala type `Array[Byte]` is a `BLOB` but postgres doesn't support it and the database returns the error:

```
[42704] ERROR: type "blob" does not exist
```

It could be that I am forgetting something in my table definition...

**Comment by dnvriend** [comment]
Please ignore, the issue was a wrong configured play:

```
//slick.dbs.default.driver="slick.driver.H2Driver$"
slick.dbs.default.driver = "slick.jdbc.PostgresProfile$"
```

**Comment by cvogt** [comment]
Re-opening this as a usability issue. If we could have prevented you from using the wrong JDBC driver, this wouldn't have happened.

**Comment by szeiger** [comment]
I don't think this should be labeled with "effort: easy" unless you already know how to implement it (and it would indeed be easy). I don't think there's a realistic solution for this problem at all. In the general case you don't know the JDBC driver, and if you do know it you still cannot know whether it matches the Slick profile or the database.

**Comment by cvogt** [comment]
agreed, wasn't intended to be marked as `easy`

One solution I can think of is pre-configuring drivers with their "default" jdbc drivers but allowing people to override that. WDYT?

**Comment by pettyjamesm** [comment]
Let me cut in. I think "solving" this issue is only possible (let alone "easy") if slick is actually under active development.

**Comment by cvogt** [comment]
@pettyjamesm We haven't been doing a great job of getting new versions out quickly. We need to get better at that and I think tooling needs to be part of the solution. Especially the dead lock issue has been frustrating, I agree.
